src/publish_control.cpp

```diff
@@ -114,6 +133,481 @@ void PublishControl::callback_veh_speed(const pacmod_msgs::VehicleSpeedRpt::Cons
   speed_mutex.unlock();
 }
 
+void PublishControl::callback_shift_rpt(const pacmod_msgs::SystemRptInt::ConstPtr& msg)
+{
+  shift_mutex.lock();
```

Comment:
Is there a need of mutex for every callback function? Where are the possible racing conditions?

Reply:
All this code was copied from `publish_control_board_rev3.cpp`, I'll refactor this in a later PR so that it's easier to follow the changes in case any bugs are introduced.

Reply:
ok